charms/tensorboard-controller/tests/integration/utils.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+
+import tenacity
+from lightkube import Client
+from lightkube.core.resource import Resource
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+
+async def setup_istio(ops_test: OpsTest, istio_gateway: str, istio_pilot: str):
+    """Deploy Istio Ingress Gateway and Istio Pilot."""
+    await ops_test.model.deploy(
+        entity_url="istio-gateway",
+        application_name=istio_gateway,
+        channel="latest/edge",
+        config={"kind": "ingress"},
+        trust=True,
+    )
+    await ops_test.model.deploy(
+        istio_pilot,
+        channel="latest/edge",
+        config={"default-gateway": "test-gateway"},
+        trust=True,
+    )
+    await ops_test.model.add_relation(istio_pilot, istio_gateway)
+
+    await ops_test.model.wait_for_idle(
+        apps=[istio_pilot, istio_gateway],
+        status="active",
+        raise_on_blocked=False,
+        timeout=60 * 20,
+    )
+
+
+async def cleanup_istio(ops_test: OpsTest, istio_gateway: str, istio_pilot: str):
+    """Remove Istio Ingress Gateway and Istio Pilot applications."""
+    for app in [istio_pilot, istio_gateway]:
+        await remove_application(ops_test, app)
```

Comment:
Is this used anywhere?

Reply:
No, it's not, I was hoping that it would go unnoticed and be left there, until removing istio actually works and we can integrate this as a cleanup step in the tests 😇

Reply:
haha ok sounds good